src/Illuminate/Cache/PhpRedisLock.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Redis\Connections\PhpRedisConnection;
+use Redis;
+use UnexpectedValueException;
+
+class PhpRedisLock extends RedisLock
+{
+    public function __construct(PhpRedisConnection $redis, string $name, int $seconds, ?string $owner = null)
+    {
+        parent::__construct($redis, $name, $seconds, $owner);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function release()
+    {
+        return (bool) $this->redis->eval(
+            LuaScripts::releaseLock(),
+            1,
+            $this->name,
+            $this->serializedAndCompressedOwner()
+        );
+    }
+
+    protected function serializedAndCompressedOwner(): string
+    {
+        $client = $this->redis->client();
+
+        /* If a serialization mode such as "php" or "igbinary" and/or a
+         * compression mode such as "lzf" or "zstd" is enabled, the owner
+         * must be serialized and/or compressed by us, because phpredis does
+         * not do this for the eval command.
+         *
+         * Name must not be modified!
+         */
+        $owner = $client->_serialize($this->owner);
```

Comment:
So adding this call will not break any existing locks that were created before this change? I could imagine a scenario where locks exist - application is deployed with this changed - will those locks still be able to be regathered and released?

Reply:
Yes they will be. If phpredis has serialization disabled, this will return the same value as passed down. I wrote a test for this case, which passes. (See: https://github.com/phpredis/phpredis/blob/develop/library.c#L3029)

Reply:
Nothing changes in regards of the redlock algorithm. The same eval call is done with the same lua script executed. This change is all about modifying the passed arguments to the eval call the same way as phpredis did on the initial lock set command.